tests/api/token/temporary.test.js

```diff
@@ -15,6 +15,22 @@ jest.mock("@lib/integration/dbConnector", () => {
   return jest.fn(() => mockClient);
 });
 
+let IsGCNotifyServiceAvailable = true;
+
+const mockSendEmail = {
+  sendEmail: jest.fn(() => {
+    if (IsGCNotifyServiceAvailable) {
+      return Promise.resolve();
+    } else {
+      throw new Error("Could not send temporary token by email.");
```

Comment:
If you throw an error then it needs to be within a Promise.reject() https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject. Or you could just use an async function here. Either works. I do like how you set this up though. Very DRY well done !